commit 882d440da2b2d88bb930218f21af46dec9240649
Author: Valentin David <valentin.david@canonical.com>
Date:   Tue Nov 30 11:15:47 2021 +0100

    Use initrd target from the beginning
    
    Also remove a copy of units to `/etc/systemd/system` as this directory
    does not exist and the copy did not seem to be useful.

diff --git a/factory/etc/systemd/system/default.target b/factory/etc/systemd/system/default.target
deleted file mode 120000
index 99121c7..0000000
--- a/factory/etc/systemd/system/default.target
+++ /dev/null
@@ -1 +0,0 @@
-/lib/systemd/system/basic.target
\ No newline at end of file
diff --git a/factory/usr/lib/systemd/system/populate-writable.service b/factory/usr/lib/systemd/system/populate-writable.service
index d88f58b..d234fcc 100644
--- a/factory/usr/lib/systemd/system/populate-writable.service
+++ b/factory/usr/lib/systemd/system/populate-writable.service
@@ -23,4 +23,3 @@ ExecStart=/usr/bin/touch /sysroot/run/fstab
 ExecStart=/usr/bin/chroot /sysroot /usr/lib/core/handle-writable-paths / /etc/system-image/writable-paths /run/fstab
 # TODO:UC20: move the-modeenv implementation to snap-bootstrap too
 ExecStart=/usr/lib/the-modeenv
-ExecStartPost=/usr/bin/systemctl --no-block isolate initrd.target
\ No newline at end of file
diff --git a/factory/usr/lib/the-modeenv b/factory/usr/lib/the-modeenv
index ee424bc..e2426a5 100755
--- a/factory/usr/lib/the-modeenv
+++ b/factory/usr/lib/the-modeenv
@@ -20,4 +20,3 @@ if grep -q snapd_recovery_mode=run /proc/cmdline; then
 fi
 mount -o bind /run/image.fstab /sysroot/etc/fstab
 umount -lf /sysroot/run
-cp /run/systemd/transient/*.mount /etc/systemd/system
diff --git a/vendor/systemd/debian/patches/series b/vendor/systemd/debian/patches/series
index a3cef58..99146f4 100644
--- a/vendor/systemd/debian/patches/series
+++ b/vendor/systemd/debian/patches/series
@@ -1,4 +1,3 @@
-ubuntu-core-initramfs-fix-default.target
 ubuntu-core-initramfs-clock-util-read-timestamp-from-usr-lib-clock-epoch.patch
 pid1-by-default-make-user-units-inherit-their-umask-from-.patch
 userdbctl-drop-redundant-user-name-validity-check.patch
diff --git a/vendor/systemd/debian/patches/ubuntu-core-initramfs-fix-default.target b/vendor/systemd/debian/patches/ubuntu-core-initramfs-fix-default.target
deleted file mode 100644
index a249fa2..0000000
--- a/vendor/systemd/debian/patches/ubuntu-core-initramfs-fix-default.target
+++ /dev/null
@@ -1,14 +0,0 @@
-Index: systemd/src/core/main.c
-===================================================================
---- systemd.orig/src/core/main.c
-+++ systemd/src/core/main.c
-@@ -2005,7 +2005,8 @@ static int do_queue_default_job(
-         if (arg_default_unit)
-                 default_unit = arg_default_unit;
-         else if (in_initrd())
--                default_unit = SPECIAL_INITRD_TARGET;
-+                // ubuntu-core-initramfs needs indirection target
-+                default_unit = SPECIAL_DEFAULT_TARGET;
-         else
-                 default_unit = SPECIAL_DEFAULT_TARGET;
- 
